core/server/master/src/test/java/alluxio/master/file/RpcContextTest.java

```diff
@@ -34,11 +35,12 @@
 public final class RpcContextTest {
   private BlockDeletionContext mMockBDC = mock(BlockDeletionContext.class);
   private JournalContext mMockJC = mock(JournalContext.class);
+  private OperationContext mMockOC = mock(OperationContext.class);
   private RpcContext mRpcContext;
 
   @Before
   public void before() {
-    mRpcContext = new RpcContext(mMockBDC, mMockJC);
+    mRpcContext = new RpcContext(mMockBDC, mMockJC, mMockOC);
   }
```

Comment:
Would like to see some test coverage on throwing for cancelled trackers.

Reply:
I can't see a clear way to test cancellation while the rpc is in progress, aka the loop break logic. I'll try adding some validation for the rest.

Reply:
I think what I would do is use Mockito's `Spy` class and wrap in conjunction with some locks to allow for the control of entering/exiting methods. See the `UfsStatusCacheTest` for an example. I'll paste some code from one of the tests inline here so you can see what I mean.


```java
createUfsDirs("dir0/dir1");
    ExecutorService executor =
        new ThreadPoolExecutor(1, 1, 1, TimeUnit.MINUTES, new SynchronousQueue<>());
    mCache = new UfsStatusCache(executor);
    mCache = Mockito.spy(mCache);
    Lock l = new ReentrantLock();
    l.lock();
    doAnswer((invocation) -> {
      try {
        l.lock();
        return invocation.callRealMethod();
      } finally {
        l.unlock();
      }
    }).when(mCache).getChildrenIfAbsent(any(AlluxioURI.class), any(MountTable.class));
    assertNotNull(mCache.prefetchChildren(new AlluxioURI("/dir0"), mMountTable));
    assertNull(mCache.prefetchChildren(new AlluxioURI("/dir0"), mMountTable)); // rejected
    assertNull(mCache.prefetchChildren(new AlluxioURI("/dir0"), mMountTable)); // rejected
    assertNull(mCache.prefetchChildren(new AlluxioURI("/dir0"), mMountTable)); // rejected
    assertNull(mCache.prefetchChildren(new AlluxioURI("/dir0"), mMountTable)); // rejected
    l.unlock();
    Collection<UfsStatus> statuses =
        mCache.fetchChildrenIfAbsent(new AlluxioURI("/dir0"), mMountTable, false);
    assertEquals(1, statuses.size());
    statuses.forEach(s -> assertEquals("dir1", s.getName()));
```

Reply:
I went for a simpler UT and think going beyond that will require an integration test with mocking which is not feasible.